smsapi/Api/Action/Base.cs

```diff
@@ -42,6 +43,30 @@ public T Execute()
             return response;
         }
 
+        public async Task<T> ExecuteAsync()
+        {
+            Validate();
+
+            T response;
+            Stream data = await proxy.ExecuteAsync(Uri(), Values(), Files(), Method);
+
+            HandleError(data);
+
+            try

```

Comment:
This is same as in sync method, please extract this part somewhere.

Reply:
89dc327